tests/integration/test_http_failover/test.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+from helpers.client import QueryRuntimeException
+from helpers.test_tools import exec_query_with_retry
+from helpers.test_tools import assert_eq_with_retry
+
+
+cluster = ClickHouseCluster(__file__)
+
+valid_ipv4 = "10.5.172.11"
+wrong_ipv6 = "2001:3984:3989::1:1118"
+
+# Destination node
+node1 = cluster.add_instance(
+    "node1",
+    with_zookeeper=True,
+    ipv4_address=valid_ipv4,
+)
+# Source node
+node2 = cluster.add_instance(
+    "node2",
+    with_zookeeper=True,
+    ipv6_address="2001:3984:3989::1:1219",
+)
+
+
+# Testing different scenarios, when the http endpoint have several ips.
+def _prepare(node):
+    node.exec_in_container(
```

Comment:
It makes sense to move this to a method like `node.add_static_dns_entry(host, ip)`.